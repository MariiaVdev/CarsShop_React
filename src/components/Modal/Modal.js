import styles from './Modal.module.scss';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { setIsOpenModalAC } from '../../store/modal/actionCreators'

const Modal = ({ header, closeButton, text, actions }) => {
    const isOpenModal = useSelector(store => store.modal.isOpenModal);
    const dispatch = useDispatch();

    if (!isOpenModal) {
        return null;
    }
    const closeModal = () => { dispatch(setIsOpenModalAC(false)) };
    return (
        <div className={styles.modal}>

            <div className={styles.modalBackground} onClick={closeModal}></div>

            <div className={styles.modalMainContainer}>
                <div className={styles.modalHeader}>
                    <p className={styles.modalHeaderText}>{header}</p>
                    {closeButton && <button className={styles.modalClose} onClick={closeModal}></button>}
                </div>
                <div className={styles.modalContentWrapper}>
                    <p>{text.name}</p>
                    <p>Are you ready to confirm your choice?</p>

                </div>
                <div className={styles.modalButtonWrapper}>
                    {actions}
                </div>
            </div>
        </div>

    )
}

Modal.propTypes = {
    header: PropTypes.string.isRequired,
    closeButton: PropTypes.bool.isRequired,
    text: PropTypes.shape({
        code: PropTypes.number,
        name: PropTypes.string
    }),
    actions: PropTypes.node.isRequired

}

Modal.defaultProps = {
    user: {
        code: 3455,
        name: 'LEXUS'
    }

}

export default Modal;