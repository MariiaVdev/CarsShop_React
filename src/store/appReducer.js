import {combineReducers} from "redux";
import modalReducer from "./modal/modalReducer";
import carsReducer from './cars/carsReducer';
import cardReducer from './card/cardReducer';
import cartReducer from './cart/cartReducer';
import formReducer from "./form/formReducer";





const appReducer = combineReducers({
    modal: modalReducer,
    cars: carsReducer,
    card: cardReducer,
    cart: cartReducer,
    form: formReducer
})


export default appReducer;