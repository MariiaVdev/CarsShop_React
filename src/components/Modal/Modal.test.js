import Modal from './Modal';
import { render, screen, fireEvent } from '@testing-library/react';
import renderWithProviders from '../../utils/utilsModal/utilsModalTest';
import { Provider } from 'react-redux';
import store from '../../store/index'

const Component = () => {
    return (
        <Provider store={store}>
            <Modal header={'header'} closeButton={true} actions={'<button>OK</button>'} />
        </Provider>
    )

}

describe('Modal renders', () => {
    test('should Modal no render without isOpen state', () => {
        const { asFragment } = render(<Component />);
        expect(asFragment()).toMatchSnapshot();
    })
    test('should Modal render when state isOpen set true', () => {
        const { asFragment } = renderWithProviders(<Modal header={'header'} closeButton={true} actions={'<button>OK</button>'} />);
        expect(asFragment()).toMatchSnapshot();

    })
    test('should Modal render without close button', () => {
        const { asFragment } = renderWithProviders(<Modal header={'header'} closeButton={false} actions={'<button>OK</button>'} />);
        expect(asFragment()).toMatchSnapshot();

    })
    test('should Modal render with close button', () => {
        const { asFragment } = renderWithProviders(<Modal header={'header'} closeButton={true} actions={'<button>OK</button>'} />);
        expect(asFragment()).toMatchSnapshot();

    })
    test('should Modal render with default text', () => {
        const { asFragment } = renderWithProviders(<Modal header={'header'} closeButton={true} actions={'<button>OK</button>'} />);
        expect(asFragment()).toMatchSnapshot();

    })
    test('should Modal render with custom text', () => {
        const { asFragment } = renderWithProviders(<Modal header={'test header'} closeButton={true} text={{ code: 1234, name: 'test' }} actions={'<button>TEST</button>'} />);
        expect(asFragment()).toMatchSnapshot();

    })
})
describe('Modal close', () => {
    test('should Modal background close work', () => {
        renderWithProviders(<Modal header={'header'} closeButton={true} actions={'<button>OK</button>'} />);
        const bgd = screen.getByTestId("Modal-bgd");
        expect(screen.getByTestId("Modal-bgd")).toBeInTheDocument();
        fireEvent.click(bgd)
        expect(screen.queryByTestId("Modal-bgd")).not.toBeInTheDocument();
    })

})
