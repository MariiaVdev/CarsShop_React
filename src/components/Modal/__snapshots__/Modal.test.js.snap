// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders should Modal no render without isOpen state 1`] = `<DocumentFragment />`;

exports[`Modal renders should Modal render when state isOpen set true 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalBackground"
      data-testid="Modal-bgd"
    />
    <div
      class="modalMainContainer"
    >
      <div
        class="modalHeader"
      >
        <p
          class="modalHeaderText"
        >
          header
        </p>
        <button
          class="modalClose"
        />
      </div>
      <div
        class="modalContentWrapper"
      >
        <p>
          LEXUS
        </p>
        <p>
          Are you ready to confirm your choice?
        </p>
      </div>
      <div
        class="modalButtonWrapper"
      >
        &lt;button&gt;OK&lt;/button&gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal renders should Modal render with close button 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalBackground"
      data-testid="Modal-bgd"
    />
    <div
      class="modalMainContainer"
    >
      <div
        class="modalHeader"
      >
        <p
          class="modalHeaderText"
        >
          header
        </p>
        <button
          class="modalClose"
        />
      </div>
      <div
        class="modalContentWrapper"
      >
        <p>
          LEXUS
        </p>
        <p>
          Are you ready to confirm your choice?
        </p>
      </div>
      <div
        class="modalButtonWrapper"
      >
        &lt;button&gt;OK&lt;/button&gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal renders should Modal render with custom text 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalBackground"
      data-testid="Modal-bgd"
    />
    <div
      class="modalMainContainer"
    >
      <div
        class="modalHeader"
      >
        <p
          class="modalHeaderText"
        >
          test header
        </p>
        <button
          class="modalClose"
        />
      </div>
      <div
        class="modalContentWrapper"
      >
        <p>
          test
        </p>
        <p>
          Are you ready to confirm your choice?
        </p>
      </div>
      <div
        class="modalButtonWrapper"
      >
        &lt;button&gt;TEST&lt;/button&gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal renders should Modal render with default text 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalBackground"
      data-testid="Modal-bgd"
    />
    <div
      class="modalMainContainer"
    >
      <div
        class="modalHeader"
      >
        <p
          class="modalHeaderText"
        >
          header
        </p>
        <button
          class="modalClose"
        />
      </div>
      <div
        class="modalContentWrapper"
      >
        <p>
          LEXUS
        </p>
        <p>
          Are you ready to confirm your choice?
        </p>
      </div>
      <div
        class="modalButtonWrapper"
      >
        &lt;button&gt;OK&lt;/button&gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal renders should Modal render without close button 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalBackground"
      data-testid="Modal-bgd"
    />
    <div
      class="modalMainContainer"
    >
      <div
        class="modalHeader"
      >
        <p
          class="modalHeaderText"
        >
          header
        </p>
      </div>
      <div
        class="modalContentWrapper"
      >
        <p>
          LEXUS
        </p>
        <p>
          Are you ready to confirm your choice?
        </p>
      </div>
      <div
        class="modalButtonWrapper"
      >
        &lt;button&gt;OK&lt;/button&gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;
